File: SampleApp/Objective-C/SampleApp/Interstitial/InterstitialViewController.m
//
//  InterstitialViewController.m
//  SampleApp
//
//  Fluct SDK
//  Copyright (c) 2014 fluct, Inc. All rights reserved.
//

#import "InterstitialViewController.h"

@import FluctSDK;

@interface InterstitialViewController () <FSSInterstitialViewDelegate>

@property (nonatomic) FSSInterstitialView *interstitialView;
@end

@implementation InterstitialViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil {
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (IBAction)showInterstitial:(id)sender {
    self.interstitialView = [[FSSInterstitialView alloc] init];
    [self.interstitialView setMediaID:@"0000000108"];
    self.interstitialView.delegate = self;
    [self.interstitialView showInterstitialAd];
    // 背景色を設定してインタースティシャル広告を表示する場合
    // [self.interstitialView showInterstitialAdWithHexColor: @"#FF0000"];
}

- (void)interstitialView:(FSSInterstitialView *)interstitialView callbackType:(FSSInterstitialViewCallbackType)callbackType {
    NSLog(@"Interstitial callback : %ld", (long)callbackType);

    switch (callbackType) {
    case FSSInterstitialViewCallbackTypeShow:
        NSLog(@"表示されました");
        break;
    case FSSInterstitialViewCallbackTypeTap:
        NSLog(@"タップされました");
        break;
    case FSSInterstitialViewCallbackTypeClose:
        NSLog(@"閉じました");
        break;
    case FSSInterstitialViewCallbackTypeCancel:
        NSLog(@"キャンセルされました");
        break;
    case FSSInterstitialViewCallbackTypeOffline:
        NSLog(@"圏外です");
        break;
    case FSSInterstitialViewCallbackTypeMediaIDError:
        NSLog(@"メディアIDが不正な値です");
        break;
    case FSSInterstitialViewCallbackTypeNoConfig:
        NSLog(@"メディアIDに設定されていません");
        break;
    case FSSInterstitialViewCallbackTypeSizeError:
        NSLog(@"表示する端末のサイズより広告が大きいです");
        break;
    case FSSInterstitialViewCallbackTypeGetConfigError:
        NSLog(@"広告情報が取得出来ませんでした");
        break;
    case FSSInterstitialViewCallbackTypeOtherError:
        NSLog(@"その他のエラーです");
        break;
    default:
        break;
    }
}

@end

File: SampleApp/Swift/SDK/SampleApp-Swift/Interstitial/InterstitialViewController.swift
//
//  InterstitialViewController.swift
//  SampleApp-Swift
//
//  Fluct SDK
//  Copyright © 2016年 fluct, Inc. All rights reserved.
//

import UIKit
import FluctSDK

class InterstitialViewController: UIViewController, FSSInterstitialViewDelegate {

    var interstitialView: FSSInterstitialView? = nil

    override func viewDidLoad() {
        super.viewDidLoad()
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }

    @IBAction func showInterstitial(_ sender: AnyObject) {
        interstitialView = FSSInterstitialView(mediaID: "0000000108")
        interstitialView?.delegate = self
        interstitialView?.showInterstitialAd()
        // 背景色を設定してインタースティシャル広告を表示する場合
        // interstitialView?.showInterstitialAdWithHexColor("#FF0000")
    }

    func interstitialView(_ interstitialView: FSSInterstitialView!, callbackType: FSSInterstitialViewCallbackType) {
        print(callbackType)
        switch (callbackType) {
        case .show:
            print("表示しました")
        case .tap:
            print("タップしました")
        case .close:
            print("閉じました")
        case .cancel:
            print("キャンセルされました")
        case .offline:
            print("圏外です")
        case .mediaIDError:
            print("メディアIDが不正な値です")
        case .noConfig:
            print("メディアIDに設定されていません")
        case .sizeError:
            print("表示する端末のサイズより広告が大きいです")
        case .getConfigError:
            print("広告情報が取得出来ませんでした")
        case .otherError:
            print("その他のエラーです")
        @unknown default:
            break
        }
    }

    // 画面の回転時にインタースティシャルを非表示にする
    override func willRotate(to toInterfaceOrientation: UIInterfaceOrientation, duration: TimeInterval) {
        interstitialView?.dismissInterstitialAd()
    }

}
